### ISSUE TITLE
Cluster creation fails when custom VPC and/or subnets are supplied

### ISSUE THREAD

**Issue by pramodvkrishna** [weaveworks/eksctl#1844]
**What happened?**
We are using custom VPC and IAM roles for creating cluster and nodegroups. I get the below error 

```
[ℹ]  eksctl version 0.14.0-rc.0
[ℹ]  using region eu-west-1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1487ba0]

goroutine 1 [running]:
github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5.NoAccess(...)
        github.com/weaveworks/eksctl@/pkg/apis/eksctl.io/v1alpha5/validation.go:125
github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5.(*ClusterConfig).ValidateClusterEndpointConfig(0xc0003abd90, 0xc000683b60, 0x0)
        github.com/weaveworks/eksctl@/pkg/apis/eksctl.io/v1alpha5/validation.go:114 +0x40
github.com/weaveworks/eksctl/pkg/ctl/create.doCreateCluster(0xc0000ea000, 0xc0002beb40, 0xc0000ea100, 0x0, 0x0)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:112 +0x168
github.com/weaveworks/eksctl/pkg/ctl/create.createClusterCmd.func1(0xc000363180, 0xc0003bd320, 0x0, 0x2, 0x0, 0x0)
        github.com/weaveworks/eksctl@/pkg/ctl/create/cluster.go:36 +0x9b
github.com/spf13/cobra.(*Command).execute(0xc000363180, 0xc0003bd2e0, 0x2, 0x2, 0xc000363180, 0xc0003bd2e0)
        github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc000362c80, 0xc0000fd940, 0x1, 0x1)
        github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
        github.com/weaveworks/eksctl@/cmd/eksctl/main.go:79 +0x407
```

**What you expected to happen?**
I was expecting to see 
```
[✔]  using existing VPC (vpc-0bed65ff4ede5a176) and subnets (private:[subnet-096889dd6eed250cf subnet-0b026c7e6f28f90c7 subnet-0e8a1ffa8d5256e93] public:[])
```

**How to reproduce it?**
1. Create cluster-poc.yml

```
$ cat cluster-poc.yml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: poc
  region: eu-west-1

vpc:
  subnets:
    private:
      eu-west-1a:
          id: "subnet-096889dd6eed250cf"
          cidr: "100.65.0.0/18"
      eu-west-1b:
          id: "subnet-0b026c7e6f28f90c7"
          cidr: "100.65.64.0/18"
      eu-west-1c:
          id: "subnet-0e8a1ffa8d5256e93"
          cidr: "100.65.128.0/18"

iam:
  serviceRoleARN: "arn:aws:iam::<acct_no>:role/poc-eks-cluster-service-role"

nodeGroups:
  - name: poc-ng

    instanceType: t3.medium
    desiredCapacity: 2
    iam:
      instanceProfileARN: "arn:aws:iam::<acct_no>:instance-profile/poc-eks-nodes-default"
      instanceRoleARN: "arn:aws:iam::<acct_no>:role/poc-eks-nodes-default"
    privateNetworking: true
    securityGroups:
      withShared: true
      withLocal: true
      attachIDs: ['sg-055e32c1064cd5125']
```
2. Run the command

```
eksctl create cluster -f cluster-poc.yml
```
**Anything else we need to know?**
OS="Ubuntu 18.04.3 LTS"
We have a regulated AWS account which doesn't permit IAM roles to be created without IAM permissions boundary set. Our previous attempt to create cluster using version 0.13.0 failed with the error below which has been fixed in version 0.14.0
```
[✖]  AWS::IAM::Role/FargatePodExecutionRole: CREATE_FAILED – "API: iam:CreateRole User: arn:aws:sts::<acct_no>:assumed-role/developer/<user> is not authorized to perform: iam:CreateRole on resource: arn:aws:iam::<acct_no>:role/eksctl-poc-cluster-FargatePodExecutionRole-TZE67RWT8KVI"
```
**Versions**
```
$ eksctl version
 ```
0.14.0-rc.0
```
$ kubectl version
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"be3d344ed06bff7a4fc60656200a93c74f31f9a4", GitTreeState:"clean", BuildDate:"2020-02-11T19:34:02Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-c0eccc", GitCommit:"c0eccca51d7500bb03b2f163dd8d534ffeb2f7a2", GitTreeState:"clean", BuildDate:"2019-12-22T23:14:11Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by Surian** [comment]
As a workaround, you can add:

**Comment by Surian** [comment]
A workaround is to explicitly set cluster endpoints in the config file:
```
vpc:
  clusterEndpoints:
    publicAccess: true/false
    privateAccess: true/false
```

**Comment by pramodvkrishna** [comment]
@Surian Thank you ! It works. I can use this workaround until the issue is fixed.

**Comment by sayboras** [comment]
@pramodvkrishna the issue is fixed in 0.14.0. Please help to close this issue. https://github.com/weaveworks/eksctl/pull/1855
